build/tutorials/tutorials-contest/hardhat-fork/README.md

```diff
@@ -0,0 +1,574 @@
+    * @param deadline Not relevant for avalanche, just pass timestamp that is in the future
+    */
+    function swap(uint256 amountOutMin, address[] calldata path, address pair, uint256 amountIn, uint256 deadline) external {
+        // We transfer the wavax from the user (msg.sender) to this contract.
+        IERC20(wavax).transferFrom(msg.sender, address(this), amountIn);
+        // We approve the router as a spender for our Wavax.
+        IERC20(wavax).approve(router, amountIn);
+        // We do te swap using the router.
+        IRouter(router).swapExactTokensForTokens(amountIn, amountOutMin, path, msg.sender, deadline);
+    }
+}
+```
+
+### Hardhat configuration
+
+As you can see we use some external contract (Pangolin router). Meaning that if you want to test this code... you'll
+have to mock/recreate this router and all the contracts that this router use ... Kinda annoying right ?
+
+Thanks to Hardhat, we can make our life easier and skip it altogether.
+
+First we need to configure Hardhat. If you look in `hardhat.config.ts` you'll see this :
+
+```ts
+const config: HardhatUserConfig = {
+    defaultNetwork: "hardhat",
+    solidity: "0.8.4",
+    networks: {
+        hardhat: {
+            chainId: 43114,
+            gasPrice: 225000000000,
+            forking: {
+                url: "https://api.avax.network/ext/bc/C/rpc",
+                enabled: true,
+            },
+        },
+        fuji: {
+            chainId: 43113,
+            gasPrice: 225000000000,
+            url: "https://api.avax-test.network/ext/bc/C/rpc",
+            accounts: [
+                PK_TEST
+            ]
+        }
+    },
+    typechain: {
+        outDir: "typechain",
+        target: "ethers-v5",
+    },
+};
+```
+
+The most interesting portion of code here is the `network` part. That's where you configure networks that you want to
+use with your project. As you can see above we have defined two networks for this tutorial:
+
+- `hardhat`, which is also the `defaultNetwork`.
+- `fuji`, which is pointing to fuji testnet.
+
+Note that you can put mutliple network definition, one of it is considered as the 'default' one. Meaning that when you
+are using
+`npx hardhat test`, it'll use the default network. If you want to run the test on another network than the default, you
+can use this variation of the command :
+`npx hardhat test --network fuji`
+
+Now let's focus on the `hardhat` one .
+
+```ts
+hardhat: {
+    chainId: 43114,
+        gasPrice
+:
+    225000000000,
+        forking
+:
+    {
+        url: "https://api.avax.network/ext/bc/C/rpc",
+            enabled
+    :
+        true,
+            blockNumber
+    :
+        2975762
+    }
+,
+}
+,
+```
+
+- `chainId` is set with to the mainnet value, as seen
+  [here](https://docs.avax.network/build/avalanchego-apis/contract-chain-c-chain-api).
+- `gasPrice` is -currently- a fixed value on avalanche and set at 225 nAVAX, stay tuned for the next release and dynamic
```

Comment:
This is no longer true, can you please accommodate dynamic gas price?

Reply:
Done